samples/attestation/common/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ target_link_libraries(
          #
          # Alternatively we could use a CMake OBJECT library, but that
          # requires a newish version of CMake.
-         $<$<PLATFORM_ID:Linux>:-Wl,--require-defined=__oe_ecalls_table>
+         $<$<PLATFORM_ID:Linux>:-Wl,--require-defined=oe_ecalls_table>
```

Comment:
This is technically a breaking change if existing apps will not build successfully without changing their build definitions when updated to the next version of OE SDK. 

Consider adding an `__oe_ecalls_table` alias to the renamed `oe_ecalls_table` and add a deprecation warning this release, then dropping the alias in the next release.

Reply:
Good catch Simon. 
`__oe_ecalls_table` is not part of the public API, and therefore renaming it is not a breaking change.

The sample ought not use unpublished symbols in the first place. 
@mingweishih Can you find out why sample was written this way? The original reasons may no longer be valid; and there might be simpler ways to achieve whatever the sample is trying to do with the linker.

Reply:
This is addressed by #3833